src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RoutingRequestTransitDataProviderFilter.java

```diff
@@ -46,26 +44,15 @@ public RoutingRequestTransitDataProviderFilter(
     this.includePlannedCancellations = includePlannedCancellations;
     this.bannedRoutes = bannedRoutes;
     this.bannedTrips = bannedTrips;
-    boolean hasOnlyMainModeFilters = allowedTransitModes
-      .stream()
-      .noneMatch(AllowedTransitMode::hasSubMode);
 
     // It is much faster to do a lookup in an EnumSet, so we use it if we don't want to filter
     // using submodes
-    if (hasOnlyMainModeFilters) {
-      EnumSet<TransitMode> allowedMainModes = allowedTransitModes
-        .stream()
-        .map(AllowedTransitMode::getMainMode)
-        .collect(Collectors.toCollection(() -> EnumSet.noneOf(TransitMode.class)));
-      transitModeIsAllowed = (Trip trip) -> allowedMainModes.contains(trip.getMode());
```

Comment:
I would assume this makes the performance worse, not better, as we go from an EnumSet contains to iteration.

Reply:
We should test the performance for everything, but one main mode (e..g. not AIRPLANE).